api/client/client.go

```diff
@@ -370,6 +373,27 @@ func (c *Client) Dialer() ContextDialer {
 	return c.dialer
 }
 
+// WaitForConnectionReady waits for the client's grpc connection finish dialing, returning an errror
+// if the ctx is canceled or the client's gRPC connection enters an unexpected state. This can be used
+// alongside the DialInBackground client config option to wait until background dialing has completed.
+func (c *Client) WaitForConnectionReady(ctx context.Context) error {
```

Comment:
this method is only needed in tests, right?
let's unexport this method

Reply:
This could/should be used anytime the client is connecting in the background, including in other tests outside of the `api/client` package.

Reply:
wouldn't those other tests just set `DialInBackground: false` or just wait for an RPC to fail?